plugins/rule_engines/irods_rule_language/src/configuration.cpp

```diff
@@ -418,23 +424,54 @@ int hash_rules(const std::vector<std::string> &irbs, const int pid, std::string
     return 0;
 }
 
-class make_copy {
+class in_memory_rulebases {
 public:
-        make_copy(const std::vector<std::string>& _irbs, const int _pid) : irbs_(_irbs), pid_(_pid) {
-            for(auto const &irb : _irbs) {
-                boost::filesystem::copy_file(get_rule_base_path(irb), get_rule_base_path_copy(irb, _pid));
-            }
-        }
-        ~make_copy() {
-            for(auto const &irb : irbs_) {
-                boost::filesystem::remove(get_rule_base_path_copy(irb, pid_));
-            }
-        }
+  in_memory_rulebases(const std::vector<std::string>& _irods_rule_bases)
+      : irods_rule_bases_(_irods_rule_bases)
```

Comment:
Do we need to store `irods_rule_bases_` in this class at all? We are using the passed in `_irods_rule_bases` to populate the `Key`s of `rule_bases_` and never end up using `irods_rule_bases_` anywhere. Even if we did, all of its contents are stored as the `Key`s in the map, so they are still available.